src/transformers/pipelines/base.py

```diff
@@ -843,6 +843,17 @@ def __init__(
                 device = next(iter(hf_device_map.values()))
             else:
                 device = -1
+                if (
+                    is_torch_mlu_available()
+                    or is_torch_cuda_available()
+                    or is_torch_npu_available()
+                    or is_torch_xpu_available(check_device=True)
+                    or is_torch_mps_available()
+                ):
+                    logging.warning(
```

Comment:
I thin this should be `logger.warning` as in CI we have many failures 

> AttributeError: module 'transformers.utils.logging' has no attribute 'warning'


I will open a PR to fix it.